server/src/main/scala/io/delta/sharing/server/DeltaSharingService.scala

```diff
@@ -343,6 +376,62 @@ object DeltaSharingService {
     server
   }
 
+  private def checkCDFOptionsValidity(
+    startingVersion: Option[String],
+    endingVersion: Option[String],
+    startingTimestamp: Option[String],
+    endingTimestamp: Option[String]): Unit = {
+    // check if we have both version and timestamp parameters
+    if (startingVersion.isDefined && startingTimestamp.isDefined) {
+      throw DeltaCDFErrors.multipleCDFBoundary("starting")
+    }
+    if (endingVersion.isDefined && endingTimestamp.isDefined) {
+      throw DeltaCDFErrors.multipleCDFBoundary("ending")
+    }
+    if (startingVersion.isEmpty && startingTimestamp.isEmpty) {
+      throw DeltaCDFErrors.noStartVersionForCDF
+    }
+    if (startingVersion.isDefined) {
+      try {
+        startingVersion.get.toLong
```

Comment:
Should we validate timestamp?
If it gets validate elsewhere, perhaps add a comment.

Reply:
Added a comment, same as the code in our repo, it's validated in the cdc reader.